spec/system/publishers/publishers_can_send_messages_to_applicants_spec.rb

```diff
@@ -67,4 +68,75 @@
       expect(conversation.reload.messages.count).to eq(2)
     end
   end
+
+  context "when jobseeker replies to messages" do
+    let(:jobseeker) { job_application.jobseeker }
+    let!(:conversation) { create(:conversation, job_application: job_application) }
+    let!(:publisher_message) { create(:message, conversation: conversation, sender: publisher, content: "Hello from publisher") }
+    let!(:job_application_without_messages) { create(:job_application, :submitted, vacancy: vacancy, jobseeker: jobseeker) }
+
+    before do
+      login_as(jobseeker, scope: :jobseeker)
+    end
+
+    it "displays existing publisher messages to jobseeker and allows jobseeker to reply to publisher messages", :js do
+      visit jobseekers_job_application_path(job_application, tab: "messages")
+
+      expect(page).to have_text("#{publisher.given_name} #{publisher.family_name}")
+      expect(page).to have_text("Hello from publisher")
+      expect(page).to have_text("Regarding application: #{vacancy.job_title}")
+      expect(page).to have_text(publisher_message.created_at.strftime("%d %B %Y at %I:%M %p"))
+
+      expect(page).to have_link("Send message to hiring staff")
+      expect(page).to have_no_link("Print this page")
+      expect(page).to have_no_text("If a candidate responds with their pre-interview documentation")
+
+      click_link "Send message to hiring staff"
+
+      expect(page).to have_css(".trix-content")
+      expect(page).to have_button("Send message")
+      expect(page).to have_link("Cancel")
+      expect(page).to have_text("How will this message be sent?")
+
+      # cannot send empty message
+      click_button "Send message"
+
+      expect(page).to have_css("h2.govuk-error-summary__title", text: "There is a problem")
+      within(".govuk-error-summary__body") do
+        expect(page).to have_link("Please enter your message")
+      end
+
+      jobseeker_reply = "Thank you for your message. I look forward to hearing from you."
+      fill_in_trix_editor "publishers_job_application_messages_form_content", with: jobseeker_reply
+
+      click_button "Send message"
+
+      expect(page).to have_text("Message sent successfully")
+      expect(page).to have_text("Hello from publisher")
+      expect(page).to have_text(jobseeker_reply)
+      expect(page).to have_text(job_application.name.to_s)
+      expect(conversation.reload.messages.count).to eq(2)
+    end
+
+    it "shows jobseeker reply in publisher interface" do
```

Comment:
This feels  like it should be a view spec rather than a functional one

Reply:
I think that's a fair comment but I also think for the sake of a one or two tests it's probably pragmatic to keep the related tests in a single file